esmvaltool/diag_scripts/radiation_budget/radiation_budget.py

```diff
@@ -0,0 +1,468 @@
+        generated.
+    model_period : string
+        The start and end years of the model dataset.
+    obs_names : list
+        The names of variables included in the observation data.
+    obs_unit : list
+        The unit of variables included in the observation data.
+    stephens_data : list
+        Stephens observation data values.
+    stephens_error : list
+        Stephens observation data error values.
+    demory_data : list
+        Demory observation data values.
+    ceres_dataset : string
+        The name of the CERES observation data.
+    ceres_data : list
+        CERES observation data values.
+    ceres_period : string
+        The start and end years of the CERES observation data.
+
+    Returns
+    -------
+    :class:`matplotlib.figure.Figure`
+        The figure containing the plot.
+    """
+    model_minus_stephens = np.array(model_data) - np.array(stephens_data)
+    model_minus_demory = np.array(model_data) - np.array(demory_data)
+    model_minus_ceres = np.array(model_data) - np.array(ceres_data)
+
+    figure, axes = plt.subplots(figsize=(12, 8))
+    title = f"Radiation budget for {model_dataset}"
+    y_label = f"Difference between model output and observations [{obs_unit}]"
+    y_lim = (-20, 20)
+    axes.set(title=title, ylabel=y_label, ylim=y_lim)
+
+    num_x_ticks = len(obs_names)
+    x_ticks = np.arange(0, num_x_ticks * 2, 2)
+
+    bar_width = 0.5
+    opacity = 0.6
+    axes.bar(
+        x_ticks + 0.2,
+        model_minus_stephens,
+        bar_width,
+        alpha=opacity,
+        color="cornflowerblue",
+        label=f"{model_dataset} ({model_period}) - Stephens et al. (2012)",
+        yerr=stephens_error,
+    )
+    axes.bar(
+        x_ticks + 0.2 + bar_width,
+        model_minus_ceres,
+        bar_width,
+        alpha=opacity,
+        color="orange",
+        label=(f"{model_dataset} ({model_period}) - {ceres_dataset} "
+               f"({ceres_period})"),
+    )
+    axes.bar(
+        x_ticks + 0.2 + bar_width * 2,
+        model_minus_demory,
+        bar_width,
+        alpha=opacity,
+        color="darkgrey",
+        label=f"{model_dataset} ({model_period}) - Demory et al. (2014)",
+    )
+    axes.spines["bottom"].set_position(("data", 0))
+    axes.spines["top"].set_position(("data", 0))
+
+    axes.set_xticks(x_ticks + bar_width + 0.5)
+    axes.set_xticklabels(obs_names, ha="center", rotation=90, fontsize=10)
+
+    axes.legend(frameon=False, fontsize=10, loc="upper left")
+
+    return figure
+
+
+def get_provenance_record(filenames):
+    """Return a provenance record describing the plot.
+
+    Parameters
+    ----------
+    filenames : list of strings
+        The filenames containing the data used to create the plot.
+
+    Returns
+    -------
+    dictionary
+        The provenance record describing the plot.
+    """
+    record = {
+        'ancestors': filenames,
+    }
+    return record
```

Comment:
We added a bare minimum provenance record. This is all we needed to add for the recipe to run without errors. The [tutorial](https://esmvalgroup.github.io/ESMValTool_Tutorial/08-diagnostics/index.html) has a very short "Recording the provenance" section, which links to the [ESMValTool documentation](https://docs.esmvaltool.org/en/latest/community/diagnostic.html#recording-provenance), but we weren't sure on the best practice here. Is there anything else that we should add here? Would it be worth adding more details to the tutorial about this?